src/sentry/eventstore/snuba/backend.py

```diff
@@ -88,6 +95,31 @@ def get_event_by_id(self, project_id, event_id, additional_columns=None):
             return SnubaEvent(result["data"][0])
         return None
 
+    def __get_event_by_id_nodestore(self, project_id, event_id):
+        event = Event(project_id=project_id, event_id=event_id)
+        event.bind_node_data()
+
+        # Return None if there was no data in nodestore
+        if len(event.data) == 0:
+            return None
+
+        event_time = datetime.fromtimestamp(event.data["timestamp"])
+
+        # Load group_id from Snuba if not a transaction
+        if event.get_event_type() != "transaction":
+            result = snuba.raw_query(
+                selected_columns=["group_id"],
+                start=event_time,
+                end=event_time + timedelta(seconds=1),
+                filter_keys={"project_id": [project_id], "event_id": [event_id]},
+                limit=1,
+            )
+
```

Comment:
Let's assert we are returning something. Otherwise it will be hard to troubleshoot which kinds of problems we can run into when returning an incomplete event.

Reply:
Sounds good